projects/plugins/protect/src/js/data/scan/use-scan-status-query.ts

```diff
@@ -41,7 +42,45 @@ export default function useScanStatusQuery( {
 
 	return useQuery( {
 		queryKey: [ QUERY_SCAN_STATUS_KEY ],
-		queryFn: API.getScanStatus,
+		queryFn: async () => {
+			// Fetch scan status data from the API
+			const data = await API.getScanStatus();
+
+			// If the scan status is not "idle", always return the fresh API data
+			if ( data.status !== 'idle' ) {
+				return data;
+			}
+
+			// Retrieve last scan timestamp from localStorage and convert to number
+			const lastRequestedScanTimestamp = Number( localStorage.getItem( 'last_requested_scan' ) );
+
+			// If there is no stored timestamp, return the API data
+			if ( ! lastRequestedScanTimestamp ) {
+				return data;
+			}
+
+			// Check if the last scan request is more than 5 minutes old
+			const isLastRequestedScanTimestampExpired =
+				lastRequestedScanTimestamp < Date.now() - 5 * 60 * 1000;
+
+			// Convert the lastChecked date string to a Unix timestamp
+			const lastCheckedTimestamp = new Date( data.lastChecked + ' UTC' ).getTime();
+
+			// Check if the scan request is completed based on the last checked time
+			// TODO: Ensure the timestamps we are comparing are in the same timezone
```

Comment:
Related to https://github.com/Automattic/jetpack/pull/39549 I believe that the `lastChecked` value is generated using the server timezone.

Reply:
Added a comment, things tested differently for me so I think this warrants further discussion/exploration.